chatsky/llm/wrapper.py

```diff
@@ -0,0 +1,182 @@
+"""
+LLM responses.
+---------
+Wrapper around langchain.
+"""
+
+try:
+    from langchain_openai import ChatOpenAI
+    from langchain_anthropic import ChatAnthropic
+    from langchain_google_vertexai import ChatVertexAI
+    from langchain_cohere import ChatCohere
+    from langchain_mistralai import ChatMistralAI
+    from langchain_core.messages import HumanMessage, SystemMessage, AIMessage
+    from langchain_core.output_parsers import StrOutputParser
+    langchain_available = True
+except ImportError:
+    langchain_available = False
+
+import base64
+import httpx
+
+from chatsky.script.core.message import Image, Message
+from chatsky.script import Context
+from chatsky.pipeline import Pipeline
+from chatsky.llm.methods import BaseMethod
+
+from typing import Union, Callable
+
+try:
+    from deepeval.models import DeepEvalBaseLLM
+    deepeval_available = True
+except ImportError:
+    deepeval_available = False
+
+
+class LLM_API(DeepEvalBaseLLM):
+    """
+    This class acts as a wrapper for all LLMs from langchain
+    and handles message exchange between remote model and chatsky classes.
+    """
+
+    def __init__(
+        self,
+        model: ChatOpenAI,
+        system_prompt: str = "",
+    ) -> None:
+        """
+        :param model: Model object.
+        :param system_prompt: System prompt for the model.
+        """
+        self.__check_imports()
+        self.model: ChatOpenAI = model
+        self.name = ""
+        self.parser = StrOutputParser()
+        self.system_prompt = system_prompt
+
+
+    def __check_imports(self):
+        if not langchain_available:
+            raise ImportError("Langchain is not available. Please install it with `pip install chatsky[llm]`.")
+        if not deepeval_available:
+            raise ImportError("DeepEval is not available. Please install it with `pip install chatsky[llm]`.")
+        
+
+    def respond(self, history: list = [""], message_schema=None) -> Message:
+        if message_schema is None:
+            result = self.parser.invoke(self.model.invoke(history))
+            result = Message(text=result)
+        elif set(message_schema.keys()).intersection(set(Message.model_fields.keys())) != set():
+            # Case if the message_schema desribes Message structure
+            raise NotImplementedError
```

Comment:
I think `message_schema` should be `Union[None, Type[Message], Type[BaseModel]]`.

And then:
```
if None:
  Message(text=result)

if issubclass(Message):
  Message.model_validate(result)

if issubclass(BaseModel):
  Message(text=str(result))
```